src/components/CPU/BreakpointPanel.js

```diff
@@ -0,0 +1,209 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import './BreakpointPanel.css';
+import { ContextMenuTrigger } from 'react-contextmenu';
+import listeners from '../../utils/listeners';
+import BreakpointContextMenu from './BreakpointPanelContextMenu';
+import BreakpointModal from './BreakpointModal';
+import BreakpointPanelItem from './BreakpointPanelItem';
+import { hasContextMenu } from '../../utils/dom';
+import { useDebuggerContext } from '../DebuggerContext';
+
+export default function BreakpointPanel(props) {

```

Comment:
I think this looks good, but it also exposes what I don't like about functional components: large functions with lots of closures (which themselves aren't trivial), significant distance between the function name and what it *does*, and many variables defined well before their contextual relevance is clear.

I think we could use a HOC here to just manage the breakpoint list though.  It could pass as props `breakpoints, selectedRow, setSelectedRow`.  That would at least separate out the toggling and key handling, and keep the gameStatus checks separate.

-[Unknown]